packages/vite/scripts/dev.ts

```diff
@@ -0,0 +1,108 @@
+import { mkdirSync, readFileSync, rmSync, writeFileSync } from 'node:fs'
+import { type BuildOptions, context } from 'esbuild'
+import packageJSON from '../package.json'
+
+rmSync('dist', { force: true, recursive: true })
+mkdirSync('dist')
+mkdirSync('dist/node')
+writeFileSync('dist/node/index.d.ts', "export * from '../../src/node/index.ts'")
+writeFileSync(
+  'dist/node/runtime.d.ts',
+  "export * from '../../src/runtime/index.ts'",
+)
+
+const serverOptions: BuildOptions = {
+  bundle: true,
+  platform: 'node',
+  target: 'node18',
+  sourcemap: true,
+  external: [
+    ...Object.keys(packageJSON.dependencies),
+    ...Object.keys(packageJSON.peerDependencies),
+    ...Object.keys(packageJSON.optionalDependencies),
+    ...Object.keys(packageJSON.devDependencies),
+  ],
+}
+const clientOptions: BuildOptions = {
+  bundle: true,
+  platform: 'browser',
+  target: 'safari14',
+  format: 'esm',
+  sourcemap: true,
+}
+
+const watch = async (options: BuildOptions) => {
+  const ctx = await context(options)
+  await ctx.watch()
+}
+
+void watch({
```

Comment:
I think it would be nice to add a comment on each of these watch calls, like `// envConfig` so it maps with https://github.com/vitejs/vite/blob/96929269949b2d5cf92ec7673df74ef0ef300f20/packages/vite/rollup.config.ts#L186-L192 and it'll be easier to debug this in the future.